packages/parameters/tests/helpers/ssmSecureStringCdk.ts

```diff
@@ -0,0 +1,55 @@
+import {
+  Context,
+  CloudFormationCustomResourceEvent
+} from 'aws-lambda';
+import {
+  SSMClient,
+  PutParameterCommand,
+  DeleteParameterCommand
+} from '@aws-sdk/client-ssm';
+
+const client = new SSMClient({});
+
+/**
+ * Create a new SSM SecureString parameter, overwriting any existing parameter with the same name if it exists.
```

Comment:
I am not sure why we need to update parameter during the test 🤔. Which test cases require that? Or do you simply want to overwrite them in case they were not cleaned up properly before?

Reply:
CDK doesn't support creating SecureString parameter, so we use a custom resource.

The comment refers only to the fact that in this custom resource handler `event.RequestType === 'Create' || event.RequestType === 'Update'` are handled by the same code path (which creates the parameter regardless of whether it exists or not).

This stack can only be created when the test is run, and it's tore down immediately after. As such, the request should never be `Update` as we don't have a way of updating the stack.

Happy to change the `if/else` statement at L47 and have instead:
```ts
if (event.RequestType === 'Create') {
     await createResource(event);
} else if (event.RequestType === 'Delete') {
     await deleteResource(event);
} else { // This now includes event.RequestType === 'Update'
     console.error('Unknown request type', event);
     throw new Error('Unknown request type');
}
```

What do you think?

Reply:
Gotcha, I didn't know about the missing feature. The stance [from CDK team is from 2020](https://github.com/aws/aws-cdk/issues/3520) to not support secrets creation, only consumptions.

Reply:
I ended up making the change in the `if/else`. This way we can detect that the parameter is being updated and make the stack fail. Given it's not an expected behavior I prefer to have a failure in the deployment rather than a silent bug.

Thanks for pointing this out!